eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,38 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper($servicePrincipal)
+{
+    $spPassword = ""
+    if ($servicePrincipal.GetType().Name -eq 'MicrosoftGraphServicePrincipal') {
```

Comment:
Should we consider using `Get-Member` to sniff for `Secret` rather than locking down to the specific type?   Something like:

```powershell
if ((Get-Member -Name "Secret" -InputOjbect $servicePrincipal -MemberType property) -eq $null) {
```

Reply:
My thinking is that being more explicit about the type makes it clear what the underlying issue is and scopes the change to only that issue, vs. a duck-typed approach. Perhaps with the function comment above that's not necessary, and `Get-Member` is better.

Reply:
Fair enough;  I think both are valid.   My thought was, hopefully, being able to adapt without changes the next time they break the universe.  😄